### PR TITLE
Added overload to ViewComponent.Json to accept JsonSerializerSettings

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Follow-up to #2524 
Related Issue - #2506 

@dougbu @rynowak

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Core/ViewComponents/JsonViewComponentResult.cs [hunk 1]
```diff
@@ -16,9 +17,20 @@ public class JsonViewComponentResult : IViewComponentResult
         /// Initializes a new <see cref="JsonViewComponentResult"/>.
         /// </summary>
         /// <param name="value">The value to format as JSON text.</param>
-        public JsonViewComponentResult(object value)
+        public JsonViewComponentResult(object value) :
+            this(value, formatter: null)
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/ViewComponents/JsonViewComponentResult.cs]: Bizarre line break.  Move colon down with `this` as in `: this(...)`.  Same for next constructor.

File: src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponent.cs [hunk 2]
```diff
@@ -173,6 +174,18 @@ public JsonViewComponentResult Json([NotNull] object value)
         }
 
         /// <summary>
+        /// Returns a result which will render JSON text.
+        /// </summary>
+        /// <param name="value">The value to output in JSON text.</param>
+        /// <param name="serializerSettings">The <see cref="JsonSerializerSettings"/> to be used by
+        /// the formatter.</param>
+        /// <returns>A <see cref="JsonViewComponentResult"/>.</returns>
+        public JsonViewComponentResult Json([NotNull] object value, [NotNull] JsonSerializerSettings serializerSettings)
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponent.cs]: Why `[NotNull]` for `value`?  No such requirement in the `JsonViewComponentResult`

**Comment by dougbu** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.Core/ViewComponents/JsonViewComponentResult.cs [hunk 3]
```diff
@@ -17,8 +18,19 @@ public class JsonViewComponentResult : IViewComponentResult
...
         public JsonViewComponentResult(object value)
+            : this(value, formatter: null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new <see cref="JsonViewComponentResult"/>.
+        /// </summary>
+        /// <param name="value">The value to format as JSON text.</param>
+        /// <param name="serializerSettings">The <see cref="JsonSerializerSettings"/> to be used by
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.Core/ViewComponents/JsonViewComponentResult.cs]: comment about caching these settings as in `JsonResult`

**Comment by rynowak** [comment]
:shipit:

**Pull request merged**